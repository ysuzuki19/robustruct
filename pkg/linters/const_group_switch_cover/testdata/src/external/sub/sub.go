package sub

type XYZ int

const (
	X XYZ = iota
	Y
	Z
)
