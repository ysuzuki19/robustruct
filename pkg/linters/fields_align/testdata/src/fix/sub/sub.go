package sub

type SubSample struct {
	A int
	b int
	C int
}

type SubPubOnly struct {
	A int
	B int
	C int
}

func AvoidLint() {
	ss := SubSample{}
	print(ss.A, ss.b, ss.C)
	pub := SubPubOnly{}
	print(pub.A, pub.B, pub.C)
}
