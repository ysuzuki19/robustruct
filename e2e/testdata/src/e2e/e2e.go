package e2e

import "e2e/sub"

type SampleStruct struct {
	Field1 int
	Field2 string
	Field3 bool
}

// succeed all
func DefinedAlignedAll() SampleStruct {
	return SampleStruct{
		Field1: 1,
		Field2: "hello",
		Field3: true,
	}
}

// failed fields_require
func DefinedLacked() SampleStruct {
	return SampleStruct{ // want "fields 'Field2, Field3' are not initialized"
		Field1: 1,
	}
}

// failed fields_align
func DefinedUnalignedAll() SampleStruct {
	return SampleStruct{ // want "all fields of the struct must be sorted by defined order"
		Field2: "hello",
		Field1: 1,
		Field3: true,
	}
}

// ignore by packages
func IgnorePackages() SampleStruct {
	// ignore:fields_require
	// ignore:fields_align
	return SampleStruct{
		Field3: true,
		Field1: 1,
	}
}

// ignore by mod
func IgnoreMod() SampleStruct {
	// ignore:robustruct
	return SampleStruct{
		Field3: true,
		Field1: 1,
	}
}

// failed fields_require, fields_align
func DefinedLackedUnaligned() SampleStruct {
	return SampleStruct{ // want "fields 'Field2' are not initialized" "all fields of the struct must be sorted by defined order"
		Field3: true,
		Field1: 1,
	}
}

type EmptyStruct struct{}

func NewEmptyStruct() EmptyStruct {
	return EmptyStruct{}
}

type WrapSub struct {
	Sub sub.SubSample
}

func NewWrapSubCollect() WrapSub {
	return WrapSub{
		Sub: sub.SubSample{
			A: 1,
		},
	}
}

func NewWrapSubEmpty() WrapSub {
	return WrapSub{
		Sub: sub.SubSample{}, // want "fields 'A' are not initialized"
	}
}

type WrapSubPub struct {
	Sub sub.SubPubOnly
}

func NewWrapSubPubCollect() WrapSubPub {
	return WrapSubPub{
		Sub: sub.SubPubOnly{
			A: 1,
			B: 2,
		},
	}
}

func NewWrapSubPubEmpty() WrapSubPub {
	return WrapSubPub{
		Sub: sub.SubPubOnly{}, // want "fields 'A, B' are not initialized"
	}
}

type WrapSubPriv struct {
	Sub sub.SubPrivOnly
}

func NewWrapSubPrivCollect() WrapSubPriv {
	return WrapSubPriv{
		Sub: sub.SubPrivOnly{},
	}
}
