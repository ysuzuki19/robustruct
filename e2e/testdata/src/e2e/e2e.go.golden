package e2e

type SampleStruct struct {
	Field1 int
	Field2 string
	Field3 bool
}

// succeed all
func DefinedAlignedAll() SampleStruct {
	return SampleStruct{
		Field1: 1,
		Field2: "hello",
		Field3: true,
	}
}

// failed fields_require
func DefinedLacked() SampleStruct {
	return SampleStruct{ // want "fields 'Field2, Field3' are not initialized"
		Field1: 1,
		Field2: "",
		Field3: false,
	}
}

// failed fields_align
func DefinedUnalignedAll() SampleStruct {
	return SampleStruct{ 
		Field1: 1,
		Field2: "hello",
		Field3: true,
	}
}

// ignore by packages
func IgnorePackages() SampleStruct {
	// ignore:fields_require
	// ignore:fields_align
	return SampleStruct{
		Field3: true,
		Field1: 1,
	}
}

// ignore by mod
func IgnoreMod() SampleStruct {
	// ignore:robustruct
	return SampleStruct{
		Field3: true,
		Field1: 1,
	}
}

// failed fields_require, fields_align
func DefinedLackedUnaligned() SampleStruct {
	return SampleStruct{
		Field1: 1,
		Field3: true,
		Field2: "",
	}
}
